sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteDocumentsService.java

```diff
@@ -447,6 +462,185 @@ public Uni<Boolean> deleteDocument(
         .transform(__ -> true);
   }
 
+  private void checkFunctionValid(BuiltInFunctionDto function) {
+    if (function.operation().equals("$push") || function.operation().equals("$pop")) {
+      return;
+    }
+    throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_INVALID_BUILTIN_FUNCTION);
+  }
+
+  /**
+   * Executes a built-in function on a sub-path of a document. Currently only supports array push
+   * and array pop.
+   *
+   * @param namespace Namespace
+   * @param collection Collection name
+   * @param documentId The ID of the document to delete
+   * @param subPath Sub-path of the document to delete. If empty will delete the whole doc.
+   * @param functionDto The function to execute.
+   * @param context Execution content
+   * @return Flag representing if the operation was success.
+   */
+  public Uni<DocumentResponseWrapper<JsonNode>> executeBuiltInFunction(
+      Uni<Schema.CqlTable> table,
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath,
+      BuiltInFunctionDto functionDto,
+      ExecutionContext context) {
+    checkFunctionValid(functionDto);
+    final List<String> subPathProcessed = processSubDocumentPath(subPath);
+    return jsonSchemaManager
+        // Ensure that no schema is present, since we are mutating the document
+        .validateJsonDocument(table, null, true)
```

Comment:
this seems odd, of we should ensure schema is there, why just not getting the schema? on the other hand, can't we do schema check if schema is there?